CMakeLists.txt

```diff
@@ -87,6 +87,11 @@ if( "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU|(Apple)?Clang" )
   endif()
 endif()
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
I think this is a good change, but what led to it?  Was there a specific assert that you wanted to fire that wasn't?

Reply:
@pniedzielski described in the later comment, the problem with Debug build on Darwin

Since this macro is disabled, we can have green ITs locally and red on GH Actions

Reply:
By the way, it's possible that it happens because we use outdated version of BDE, but it's easier to just enable this condition explicitly

Reply:
Makes sense to me!